tests/jballerina-unit-test/src/test/resources/test-src/workers/sync-send.bal

```diff
@@ -629,6 +629,211 @@ public function testFailureForReceiveWithError() returns boolean {
     return x.w1 is E2 && x.w2 === true;
 }
 
+string appendCloneable = "";
+function testSimpleSyncSendWithCloneableExpression() {
+    string done = processWithCloneableExpression();
+    assertValueEquality("done", done);
+    assertValueEquality("w2cw2cw2cw2cw2cw1cw1cw1cw1cw1c", appendCloneable);
+}
+
+function processWithCloneableExpression() returns string {
+    worker w1Cloneable returns boolean {
+        value:Cloneable w1a = 10;
+        value:Cloneable w1b = 11;
+        value:Cloneable w1c = 12;
+        w1a -> w2Cloneable;
+        w1b ->> w2Cloneable;
+        w1c -> w2Cloneable;
+        foreach var i in 1 ... 5 {
+            appendCloneable = appendCloneable + "w1c";
+        }
+        return true;
+    }
+
+    worker w2Cloneable returns boolean {
+        value:Cloneable w2a = 15;
+        value:Cloneable w2b = 15;
+        value:Cloneable w2c = 15;
+        sleep(10);
+        foreach var i in 1 ... 5 {
+            appendCloneable = appendCloneable + "w2c";
+        }
+        w2a = <- w1Cloneable;
+        w2b = <- w1Cloneable;
+        w2c = <- w1Cloneable;
+        if (w2a is int) {
+            assertValueEquality(10, w2a);
+        } else {
+            return false;
+        }
+        if (w2b is int) {
+            assertValueEquality(11, w2b);
+        } else {
+            return false;
+        }
+        if (w2c is int) {
+            assertValueEquality(12, w2c);
+        } else {
+            return false;
+        }
+        return true;
+    }
+
+    record { boolean w1Cloneable; boolean w2Cloneable; } x = wait {w1Cloneable, w2Cloneable};
+    assertValueEquality(true, x.w1Cloneable);
+    assertValueEquality(true, x.w2Cloneable);
+    return "done";
```

Comment:
Returning and asserting this doesn't add any value right?

Reply:
Just to make sure the worker actually worked.